secret/secrets_manager_vault.go

```diff
@@ -20,20 +23,56 @@ func NewBasicSecretsManager(c SecretsManagerClient) *BasicSecretsManager {
 
 // CreateSecret creates a new secret.
 func (m *BasicSecretsManager) CreateSecret(ctx context.Context, s NamedSecret) (id string, err error) {
-	return "", errors.New("TODO: implement")
+	if err := s.Validate(); err != nil {
+		return "", err
```

Comment:
nit: can you wrap this with `errors.Wrap(err, "invalid secret")`? This is helpful mostly for tracing errors when someone calls this code.